.github/workflows/haskell.yml

```diff
@@ -11,12 +11,16 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ghc: ["8.10.7", "9.2.7", "9.6.2"]
-        cabal: ["3.10.1.0"]
-        os: [ubuntu-latest, macos-latest, windows-latest]
-        exclude:
-          - ghc: "9.2.7"
-            os: macos-latest
+        # If you edit these versions, make sure the version in the lonely macos-latest job below is updated accordingly
+        ghc: ["9.2.8", "9.6.3"]
+        cabal: ["3.10.2.1"]
+        os: [windows-latest, ubuntu-latest]
+        include:
+          # Using include, to make sure only be one macOS job, even if the matrix gets expanded later on.
+          # We want a single job, because macOS runners are scarce.
```

Comment:
```suggestion
          # Using include, to make sure there will be only one macOS job, even if the matrix gets expanded later on.
          # We want a single job, because macOS runners are scarce.
```